C#/Botkit/Microsoft.BotKit/BotkitFrameworkAdapter.cs

```diff
@@ -1,33 +1,105 @@
-// Copyright(c) Microsoft Corporation.All rights reserved.
-// Licensed under the MIT License.
-
-using Microsoft.Bot.Builder;
-using Microsoft.Bot.Connector.Authentication;
 using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
 using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Newtonsoft.Json;
 
 namespace Microsoft.BotKit
 {
     public class BotkitFrameworkAdapter : BotFrameworkAdapter
     {
+        protected readonly MicrosoftAppCredentials _credentials;
+
         /// <summary>
         /// This class extends the BotFrameworkAdapter with a few additional features to support Microsoft Teams.
         /// </summary>
         /// <param name="options"></param>
-        public BotkitFrameworkAdapter(ICredentialProvider options) : base(options)
+        /// <param name="credentials"></param>
+        public BotkitFrameworkAdapter(ICredentialProvider options, MicrosoftAppCredentials credentials) : base(options)
         {
-
+            _credentials = credentials;
         }
 
         /// <summary>
         /// Get the list of channels in a MS Teams team.
         /// Can only be called with a TurnContext that originated in a team conversation - 1:1 conversations happen _outside a team_ and thus do not contain the required information to call this API.
         /// </summary>
-        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams</param>
-        /// <returns>An array of channels in the format [{name: string, id: string}]</returns>
+        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams.</param>
+        /// <returns>An array of channels in the format [{name: string, id: string}].</returns>
         public async Task<Tuple<string, string>> GetChannels(TurnContext turnContext)
         {
-            return new Tuple<string, string>("","");
+            if(turnContext.Activity.ChannelData != null && (turnContext.Activity.ChannelData as dynamic).team != null) //TO-DO: replace 'as dynamic'
+            {
+                var token = await _credentials.GetTokenAsync(true);
+
+                var uri = $"{ turnContext.Activity.ServiceUrl }v3/teams/{ (turnContext.Activity.ChannelData as dynamic).team.id }/conversations/"; //TO-DO: replace 'as dynamic'
+
+                return await RequestUrl(token, uri);
+            }
+            else
+            {
+                return new Tuple<string, string>("", "");
+            }
+        }
+
+        /// <summary>
+        /// Perform the Get request.
+        /// </summary>
+        /// <param name="token">An authorization token.</param>
+        /// <param name="uri">Url for the Get method.</param>
+        public async Task<Tuple<string, string>> RequestUrl(string token, string uri)
```

Comment:
If we are going to update the returning type of `getChannels`, we have to update this method too